Program.cs

```diff
@@ -46,169 +48,108 @@ public static void Main()
-                    }
-                    if (snakeCol < 0)
-                    {
-                        // Console.WriteLine("The snake has gone too far left");
-                        snakeCol = gridCols - 1;
-                    }
-
-                    int a = prevLocRow.Length - 2;
-                    while (a >= 0)
-                    {
-                        prevLocRow[a + 1] = prevLocRow[a];
-                        prevLocCol[a + 1] = prevLocCol[a];
-                        a = a - 1;
-                    }
-                    prevLocRow[0] = snakeRow;
-                    prevLocCol[0] = snakeCol;
-
-                    int b = 2; // this is set as 2 as otherwise it just ignores the last part of the tail 
-                    while (b <= score)
-                    {
-                        if (snakeRow == prevLocRow[b] && snakeCol == prevLocCol[b])
-                        {
-                            gameOver = true;
-                        }
-                        b = b + 1;
-                    }
-
-                    // printing to check
-                    // Console.WriteLine("rows");
-                    // foreach (int row in prevLocRow)
-                    // {
-                    //     Console.WriteLine(row);
-                    // }
-                    // Console.WriteLine("cols");
-                    // foreach (int col in prevLocCol)
-                    // {
-                    //     Console.WriteLine(col);
-                    // }
+            // UPDATE STATE SECTION 
+            // Code to set direction based on keyboard input
+            if (Raylib.IsKeyPressed(KeyboardKey.Up)) snake.SetDirection(Snek.Direction.Up);
+            if (Raylib.IsKeyPressed(KeyboardKey.Down)) snake.SetDirection(Snek.Direction.Down);
+            if (Raylib.IsKeyPressed(KeyboardKey.Left)) snake.SetDirection(Snek.Direction.Left);
+            if (Raylib.IsKeyPressed(KeyboardKey.Right)) snake.SetDirection(Snek.Direction.Right);
 
-                }
+            // Code to move Snek and check for overlap 
+            double currentFrame = Raylib.GetTime();
+            if (!gameOver && currentFrame - lastFrameMoved >= secondsToMove)
+            {
+                lastFrameMoved = currentFrame;
+                snake.Move(gridRows, gridCols);
+                gameOver = snake.CheckOverlap(score);
+            }
 
-                // Apple eating situation 
-                if ((snakeRow == appleRow) && (snakeCol == appleCol))
-                {
-                    score += 1;
-                    Console.WriteLine("Score:" + score);
-                    secondsToMove *= speedFactor;
-                    appleRow = rng.Next(0, gridRows);
-                    appleCol = rng.Next(0, gridCols);
-                    center = CellToCenter(appleRow, appleCol, squareSize);
-                }
+            // Apple eating situation 
+            if (snake.HasEaten(apple.appleRow, apple.appleCol))
```

Comment:
Why are you passing the rows of the apple and the columns individually. Why now

`snake.HasEaten(apple)`